### PR TITLE
Add module config object for network monitor

### LINKED ISSUE #110: Reinstate network connectivity watcher in SMIBHID
Network connectivity watcher is no longer started as of the refactor of space state from HID into its own module. It isn't really needed until push messages are supported as outgoing connection will check for and establish connectivity first.

I don't want to reintroduce it until I have worked out the correct design pattern to prevent passing the wifi object down the full stack of included modules manually. Micropython doesn't support DI frameworks, but chatgpt suggested a factory approach that I am looking into.

### PR DESCRIPTION
**Pull request by sjefferson99**

### CODE REVIEW

File: smibhid/lib/hid.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
 from ulogging import uLogger
 from asyncio import get_event_loop
-from slack_api import Wrapper
```
Comment (samcorky) [inline smibhid/lib/hid.py:3]: Not sure about this kind of import. I'd rather it be explicitly relative or absolute. Will approve but need to discuss it tomorrow.

**Review by samcorky — APPROVED** [review]
Not sure about this kind of import. I'd rather it be explicitly relative or absolute. Will approve but need to discuss it tomorrow.

File: smibhid/lib/hid.py (hunk 1 above)
Comment (sjefferson99) [reply smibhid/lib/hid.py:3]: These are all relative imports? But I know I've mixed relative and partial absolutes elsewhere due to weirdness in the pico failing to resolve them. We can test some more tomorrow and work out how to resolve them if possible.

**Comment by samcorky** [comment]
I mean they should really be `from .slack_api import Wrapper` to signify
it's relative. Without the dot, it assumes that `slack_api` is a top level
module or package.

Either it should be `from smibhid.lib.slack_api import Wrapper` or `from .slack_api import Wrapper`

...


For some reason in-line code snippets aren't rendering for me

**Review by samcorky — APPROVED** [review]

**Pull request merged**